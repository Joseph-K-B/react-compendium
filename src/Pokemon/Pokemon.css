img{
    width: 100%;
    height: 100%;
}

.pokemonFigure{
    display: flex;
    flex-direction: column;
    align-items: center;
    text-align: center;
    margin: 0.5%;
    padding: 0.5%;
    height: 100%;
    width: 100%;
    border: solid #D3D0CB 0.25rem;
}