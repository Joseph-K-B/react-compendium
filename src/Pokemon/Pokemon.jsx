import './Pokemon.css';
import PropTypes from 'prop-types';

function Pokemon({ pokemon }) {
    const {
        name, 
        abilityOne, 
        abilityTwo, 
        typeOne, 
        typeTwo, 
        pic, 
        hiddenAbility 
    } = pokemon

    return(
        <figure 
            key={pokemon.id} 
            aria-label='pokemonFigure'
            className='pokemonFigure'    
        >
            <h2>{name}</h2>
            <img src={pic} alt={name} aria-label='pokemonName'/>
            <h2>Type</h2>
            <h3>{typeOne}</h3>
            <ul>
                <li>Secondary type: {typeTwo}</li>
                <li>abilitiy: {abilityOne}</li>
                <li>ability: {abilityTwo}</li>
                <li>hidden: {hiddenAbility}</li>
            </ul>
        </figure>
    )
    
}
    Pokemon.propTypes = {
        pokemons: PropTypes.arrayOf(
            PropTypes.shape({
                id: PropTypes.number.isRequired,
                name: PropTypes.string.isRequired,
                typeOne: PropTypes.string.isRequired,
                typeTwo: PropTypes.string.isRequired,
                abilityOne: PropTypes.string.isRequired,
                abilityTwo: PropTypes.string.isRequired,
                hiddenAbility: PropTypes.string.isRequired,
                pic: PropTypes.string.isRequired,
            })
        ).isRequired,
}
export default Pokemon