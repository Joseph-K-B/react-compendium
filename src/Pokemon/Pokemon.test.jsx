import { render, screen } from "@testing-library/react"
import Pokemon from "./Pokemon"


it('should render individual pokemon', () => {
    const {container} = render(<Pokemon pokemon={['pikachu']}/>);
    const figure = screen.getByLabelText(/pokemonFigure/);
    const image = screen.getByLabelText(/pokemonName/);

    expect(figure).toBeInTheDocument();
    expect(image).toBeInTheDocument();
    expect(container).toMatchSnapshot();
});