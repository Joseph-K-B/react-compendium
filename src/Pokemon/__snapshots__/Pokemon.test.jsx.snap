// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render individual pokemon 1`] = `
<div>
  <figure
    aria-label="pokemonFigure"
    class="pokemonFigure"
  >
    <h2 />
    <img
      aria-label="pokemonName"
    />
    <h2>
      Type
    </h2>
    <h3 />
    <ul>
      <li>
        Secondary type: 
      </li>
      <li>
        abilitiy: 
      </li>
      <li>
        ability: 
      </li>
      <li>
        hidden: 
      </li>
    </ul>
  </figure>
</div>
`;
